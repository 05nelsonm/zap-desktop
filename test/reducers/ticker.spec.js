import tickerReducer, {
  SET_CURRENCY,
  GET_TICKER,
  RECIEVE_TICKER
} from '../../app/reducers/ticker'

describe('reducers', () => {
  describe('tickerReducer', () => {
    it('should handle initial state', () => {
      expect(tickerReducer(undefined, {})).toMatchSnapshot()
    })

    it('should have SET_CURRENCY', () => {
      expect(SET_CURRENCY).toEqual('SET_CURRENCY')
    })

    it('should have GET_TICKER', () => {
      expect(GET_TICKER).toEqual('GET_TICKER')
    })

    it('should have RECIEVE_TICKER', () => {
      expect(RECIEVE_TICKER).toEqual('RECIEVE_TICKER')
    })

    it('should correctly setCurrency', () => {
      expect(tickerReducer(undefined, { type: SET_CURRENCY, currency: 'foo' })).toMatchSnapshot()
    })

    it('should correctly getTicker', () => {
      expect(tickerReducer(undefined, { type: GET_TICKER })).toMatchSnapshot()
    })

    it('should correctly receiveTicker', () => {
      expect(tickerReducer(undefined, { type: RECIEVE_TICKER, ticker: 'foo' })).toMatchSnapshot()
    })
  })
})
