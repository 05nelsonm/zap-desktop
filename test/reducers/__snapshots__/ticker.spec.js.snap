// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers tickerReducer should correctly getTicker 1`] = `
Object {
  "btcTicker": null,
  "crypto": "btc",
  "currency": "btc",
  "tickerLoading": true,
}
`;

exports[`reducers tickerReducer should correctly receiveTicker 1`] = `
Object {
  "btcTicker": "f",
  "crypto": "btc",
  "currency": "btc",
  "tickerLoading": false,
}
`;

exports[`reducers tickerReducer should correctly setCurrency 1`] = `
Object {
  "btcTicker": null,
  "crypto": "btc",
  "currency": "foo",
  "tickerLoading": false,
}
`;

exports[`reducers tickerReducer should handle initial state 1`] = `
Object {
  "btcTicker": null,
  "crypto": "btc",
  "currency": "btc",
  "tickerLoading": false,
}
`;
